src/components/location/location.service.ts

```diff
@@ -117,7 +104,10 @@ export class LocationService {
     // no canList check needed because all roles can list
     const results = await this.repo.list(input, session);
 
-    return await mapListResults(results, (dto) => this.secure(dto, session));
+    return {
+      ...results,
+      items: results.items.map((dto) => this.secure(dto, session)),
+    };
   }
 
   async addLocationToNode(label: string, id: ID, rel: string, locationId: ID) {
```

Comment:
@CarsonF - Also, we have extra methods here in the service like this one.  If there is going to be some sort of equivalent to these being called here for EdgeDB, then we should leave them.  But if these are also specific to Neo4j, then now could be the time to purge them as well.  The only way I can think to do this would be to remove this method here (and move any salient pieces into it's sister repo method), for example, and have all of it's callers call directly into the repo instead of here?

Thoughts appreciated.

Reply:
We will need these for EdgeDB too. Since actual GQL migrations will call these methods directly for updates, they make sense to keep. (As opposed to it needing to be a piece of another larger query).

Working on a couple type changes to help with this implementation.

Reply:
#3077

This should get you started:
```ts
const res = this.resources.getByEdgeDB(label);
const node = e.cast(res.db, e.cast(e.uuid, id));
const location = e.cast(e.Location, e.cast(e.uuid, locationId));
const query = e.update(node, () => ({
  set: {
    [rel]: { '+=': location },
  },
}));
```

Reply:
I'm sorry; I'm not following.  Is that block going to replace what is in `LocationService.addLocationToNode`?

Reply:
That's `addLocationToNode` for the edgedb repository